File: easy/2824. Count Pairs Whose Sum is Less than Target/solution.java
class Solution {
    public int countPairs(List<Integer> n, int t) {
        int ans = 0, len = n.size();
        for(int i = 0; i < len; ++i)
            for(int j = i + 1; j < len; ++j) 
                if(n.get(i) + n.get(j) < t) 
                    ++ans;
        return ans;
    }
}

File: easy/2824. Count Pairs Whose Sum is Less than Target/solution.rs
impl Solution {
    pub fn count_pairs(mut nums: Vec<i32>, target: i32) -> i32 {
        nums.sort_unstable();
        let mut left = 0;
        let mut right = nums.len() - 1;
        let mut count = 0;

        while left < right {
            if nums[left] + nums[right] < target {
                count += (right - left) as i32;
                left += 1;
            } else {
                right -= 1;
            }
        }

        count
    }
}